analyzer/src/funTest/kotlin/PubTest.kt

```diff
@@ -0,0 +1,152 @@
+            "resolve dart http dependencies correctly" {
+                val workingDir = File(projectsDirExternal, "http")
+                val packageFile = File(workingDir, "pubspec.yaml")
+                var expectedResultFile = File(projectsDirExternal, "http-expected-output.yml");
+
+                val result = createPubForExternal().resolveDependencies(listOf(packageFile))[packageFile]
+                result!!.errors.size shouldBe 0
+
+                //expectedResultFile.writeText(yamlMapper.writeValueAsString(result));
+
+                val vcsPath = vcsDir.getPathToRoot(workingDir)
+                val expectedResult = patchExpectedResult(
+                    expectedResultFile,
+                    custom = Pair("pub-project", "pub-${workingDir.name}"),
+                    definitionFilePath = "$vcsPath/pubspec.yaml",
+                    url = normalizeVcsUrl(vcsUrl),
+                    revision = vcsRevision,
+                    path = vcsPath
+                )
+
+                yamlMapper.writeValueAsString(result) shouldBe expectedResult
+            }
+
+            "Resolve dependencies for a project with flutter correctly" {
+                val workingDir = File(projectsDir, "project-with-flutter")
+                val packageFile = File(workingDir, "pubspec.yaml")
+                var expectedResultFile = File(projectsDir.parentFile, "pub-expected-output-project-with-flutter.yml");
+
+                val result = createPub().resolveDependencies(listOf(packageFile))[packageFile]
+
+                if(!result!!.errors.isEmpty()) {
+                    print(result.errors.get(0).message);
+                }
+                //result.errors.size shouldBe 0
+
+                //expectedResultFile.writeText(yamlMapper.writeValueAsString(result));
+
+                val vcsPath = vcsDir.getPathToRoot(workingDir)
+                val expectedResult = patchExpectedResult(
+                    expectedResultFile,
+                    custom = Pair("pub-project", "pub-${workingDir.name}"),
+                    definitionFilePath = "$vcsPath/pubspec.yaml",
+                    url = normalizeVcsUrl(vcsUrl),
+                    revision = vcsRevision,
+                    path = vcsPath
+                )
+
+                yamlMapper.writeValueAsString(result) shouldBe expectedResult
+
+            }
+
+            "Resolve dependencies for a project with dependencies without a static version" {
+                val workingDir = File(projectsDir, "any-version")
+                val packageFile = File(workingDir, "pubspec.yaml")
+                var expectedResultFile = File(projectsDir.parentFile, "pub-expected-output-any-version.yml");
+
+                val result = createPub().resolveDependencies(listOf(packageFile))[packageFile]
+
+                if(!result!!.errors.isEmpty()) {
+                    print(result.errors.get(0).message);
+                }
+                result.errors.size shouldBe 0
```

Comment:
This check is implicitly done by the comparison below. If you remove the check the diff show of the comparison shows the error message.

Reply:
done